pkg/changelog/entry.go

```diff
@@ -0,0 +1,146 @@
+package changelog
+
+import "time"
+
+// entry is an internal structure representing a single change in the entryList.
+// It includes the timestamp and the value of the change.
+type entry[T comparable] struct {
+	tsUnixNano int64 // timestamp of the change (nanoseconds since epoch)
```

Comment:
Replaced time.Time with int64 (UnixNano). I believe that's a safe change since it's a internal field for checking just time passing diffs. It got us ~4MB extra bytes (proctree with 32k elements in both caches).